diff --git a/source/aerial_lab/aerial_lab/tasks/direct/beetle/agents/rsl_rl_ppo_cfg.py b/source/aerial_lab/aerial_lab/tasks/direct/beetle/agents/rsl_rl_ppo_cfg.py
index 470d979..02acc94 100644
--- a/source/aerial_lab/aerial_lab/tasks/direct/beetle/agents/rsl_rl_ppo_cfg.py
+++ b/source/aerial_lab/aerial_lab/tasks/direct/beetle/agents/rsl_rl_ppo_cfg.py
@@ -33,10 +33,10 @@ class BeetlePPORunnerCfg(RslRlOnPolicyRunnerCfg):
         value_loss_coef=1.0,
         use_clipped_value_loss=True,
         clip_param=0.2,
-        entropy_coef=0.0,
+        entropy_coef=0.005,
         num_learning_epochs=5,
         num_mini_batches=4,
-        learning_rate=5.0e-4,  # default 5.0e-4
+        learning_rate=1.0e-3,  # default 5.0e-4
         schedule="adaptive",
         gamma=0.99,
         lam=0.95,
diff --git a/source/aerial_lab/aerial_lab/tasks/direct/beetle/beetle_env.py b/source/aerial_lab/aerial_lab/tasks/direct/beetle/beetle_env.py
index 784412b..51745ef 100644
--- a/source/aerial_lab/aerial_lab/tasks/direct/beetle/beetle_env.py
+++ b/source/aerial_lab/aerial_lab/tasks/direct/beetle/beetle_env.py
@@ -88,7 +88,7 @@ class BeetleEnvCfg(DirectRLEnvCfg):
     # last action (8)
     observation_space = 9 + 6 + 3 + 6 + 3 + gimbal_num + action_space
     thrust_to_torque_ratio = 0.0165
-    rotor_direction = [1, -1, -1, 1]
+    rotor_direction = [1, -1, 1, -1]  # beetle_hyper, joint urdf configuraion
     # rotor_direction = [1, 1, 1, 1]
     contact_force_threshold = 0.1
 
@@ -144,8 +144,8 @@ class BeetleEnvCfg(DirectRLEnvCfg):
     angular_to_goal_reward_scale = 2.5
 
     died_reward_scale = -1.0
-    reach_goal_reward_timeout_scale = 0.0
-    reach_goal_reward_scale = 0.5
+    reach_goal_reward_timeout_scale = 0.1
+    reach_goal_reward_scale = 0.1
 
     # # # # Noise Configuration
     noiseCfg = {
@@ -316,7 +316,7 @@ class BeetleEnv(DirectRLEnv):
                 # "angular_error_to_goal",
                 "angular_to_goal",
                 "died",
-                # "reach_goal",
+                "reach_goal",
                 # "reach_goal_timeout",
             ]
         }
@@ -651,24 +651,24 @@ class BeetleEnv(DirectRLEnv):
             torch.logical_and(ang_vel_norm < ANG_VEL_TH, lin_vel_norm < LIN_VEL_TH),
             torch.logical_and(angular_to_goal < ANG_TH, distance_to_goal < POS_TH),
         )
-        self._reach_goal_count += reach_goal.to(torch.float32) * self.reset_time_outs.to(torch.float32)
         self._reach_goal = reach_goal.to(torch.float32) * self.reset_time_outs.to(torch.float32)
+        self._reach_goal_count += self._reach_goal
         reach_goal_reward_timeout = (
-            self.reset_time_outs.to(torch.float32) * reach_goal.to(torch.float32) * self.episode_length_buf
+            self.reset_time_outs.to(torch.float32) * reach_goal.to(torch.float32) * self.max_episode_length_s
         )
         reach_goal_reward = torch.zeros_like(reach_goal_reward_timeout)
-        reach_goal_reward += reach_goal.to(torch.float32) * self.step_dt
+        reach_goal_reward += reach_goal.to(torch.float32) * self.step_dt * 5.0
         reach_goal_reward += (
-            torch.exp((POS_TH - distance_to_goal) / POS_TH) * reach_goal.to(torch.float32) * self.step_dt
+            - distance_to_goal / POS_TH * reach_goal.to(torch.float32) * self.step_dt
         )
         reach_goal_reward += (
-            torch.exp((ANG_TH - angular_to_goal) / ANG_TH) * reach_goal.to(torch.float32) * self.step_dt
+            - angular_to_goal / ANG_TH * reach_goal.to(torch.float32) * self.step_dt
         )
         reach_goal_reward += (
-            torch.exp((LIN_VEL_TH - lin_vel_norm) / LIN_VEL_TH) * reach_goal.to(torch.float32) * self.step_dt
+            - lin_vel_norm / LIN_VEL_TH * reach_goal.to(torch.float32) * self.step_dt
         )
         reach_goal_reward += (
-            torch.exp((ANG_VEL_TH - ang_vel_norm) / ANG_VEL_TH) * reach_goal.to(torch.float32) * self.step_dt
+            - ang_vel_norm / ANG_VEL_TH * reach_goal.to(torch.float32) * self.step_dt
         )
 
         # self._reach_goal_count = reach_goal.to(torch.float32) * (self._reach_goal_count + 1) * self.reset_time_outs.to(torch.float32)
@@ -704,8 +704,8 @@ class BeetleEnv(DirectRLEnv):
         #     * self.max_episode_length_s
         # )
         # reach_goal_reward = torch.clamp(reach_goal_reward, min=0.0, max=40.0)
-        # total_reward += reach_goal_reward * self.cfg.reach_goal_reward_scale
-        # rewards["reach_goal"] = reach_goal_reward
+        total_reward += reach_goal_reward * self.cfg.reach_goal_reward_scale
+        rewards["reach_goal"] = reach_goal_reward
         # rewards["reach_goal_timeout"] = reach_goal_reward_timeout
         # total_reward += reach_goal_reward_timeout * self.cfg.reach_goal_reward_timeout_scale
         # reach_goal_precise_reward = self.reset_time_outs.to(torch.float32) * reach_goal_precise.to(torch.float32) * 30.0
@@ -743,10 +743,10 @@ class BeetleEnv(DirectRLEnv):
             env_ids = self._robot._ALL_INDICES
 
         quat_sample_rate = (
-            (self.common_step_counter - 200 * 24) / self.cfg.max_curricular_steps * 4
+            (self.common_step_counter + 100 * 24) / self.cfg.max_curricular_steps * 12
         )  # start from 0.3, reach 0.8
         pos_sample_rate = (
-            (self.common_step_counter - 50 * 24) / self.cfg.max_curricular_steps * 8
+            (self.common_step_counter + 100 * 24) / self.cfg.max_curricular_steps * 12
         )  # start from 0.1, reach 0.6
         quat_sample_rate = max(quat_sample_rate, 0.0)
         pos_sample_rate = max(pos_sample_rate, 0.0)
@@ -757,8 +757,8 @@ class BeetleEnv(DirectRLEnv):
 
         quat_sample_rate = min(quat_sample_rate, 1.0)
         pos_sample_rate = min(pos_sample_rate, 1.0)
-        quat_sample_rate = 1.0
-        pos_sample_rate = 1.0
+        # quat_sample_rate = 1.0
+        # pos_sample_rate = 1.0
         self._quat_sample_rate = quat_sample_rate
         self._pos_sample_rate = pos_sample_rate
 
